### PR TITLE
Feature/vector topic model

### PR DESCRIPTION
**Pull request by BrandonSmith710**
Description: This file proposes a new method of topic modeling, called Non-Negative Matrix Factorization. The function, "topicizer" will accept a list of meeting notes, clean and vectorize the text, then return the desired number of topics; all from a minimally dependent environment.

Dependencies: sklearn.feature_extraction.text, sklearn.decomposition

Type of change:
- [ ] New feature
- [ ] This change requires a documentation update

Checklist:
- [ ] My code follows PEP8 style guide
- [ ] I have removed unnecessary print statements from my code
- [ ] I have made corresponding changes to the documentation if necessary
- [ ] My changes generate no errors
- [ ] No commented-out code
- [ ] Size of pull request kept to a minimum
- [ ] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

File: app/document_frequency.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      num_topics - Positive integer. The number of topics to present the entire
+                   text. Will repeat topics if a large number is specified
+                   but there is not enough data.
+                  
+      len_topic_repr - Positive integer. The number of words that will be used
+                       to represent each topic.
+    Returns: A list of topic representations, each is a list
+             of the most influential words for the topic.
+    '''
+
```
Comment (BrokenShell) [inline app/document_frequency.py:44]: Are all these blank lines really necessary? Also I'd reconsider most of your inline comments... eg.. If I know what an inverse document matrix is, then I also know what the code means.

Comment (BrandonSmith710) [reply app/document_frequency.py:44]: Thanks. And ok, I can definitely remove the blank lines and comments.

File: app/document_frequency.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
+from sklearn.decomposition import NMF
+from sklearn.feature_extraction.text import TfidfVectorizer
+import re
+
+# max_df allows us to ignore terms with a document frequency greater than .8
+tfidf_vect = TfidfVectorizer(max_df=.8, stop_words='english')
+
+def cleaner(text):
+    '''
```
Comment (jinjahninjah) [inline app/document_frequency.py:9]: Just swap out these single quotes for double quotes 👍

File: app/document_frequency.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    Removes extra spaces, symbols, and punctuation from string.
+    
+    Parameters: text - string
+    Returns: a lowercased version of the string
+    '''
```
Comment (jinjahninjah) [inline app/document_frequency.py:14]: Just swap out these single quotes for double quotes 👍

File: app/document_frequency.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+    text = text.replace('\n', ' ')
+    text = re.sub('[^a-z A-Z0-9]', '', text)
+    text = re.sub('[ ]{2, }', ' ', text)
+
+    return text.lower().strip()
+
+
+def topicizer(texts, num_topics = 5, len_topic_repr = 5):
+    '''
```
Comment (jinjahninjah) [inline app/document_frequency.py:25]: Just swap out these single quotes for double quotes 👍

File: app/document_frequency.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+                   a total of at least 10 unique words within the list.
+      num_topics - Positive integer. The number of topics to present the entire
+                   text. Will repeat topics if a large number is specified
+                   but there is not enough data.
+                  
+      len_topic_repr - Positive integer. The number of words that will be used
+                       to represent each topic.
+    Returns: A list of topic representations, each is a list
+             of the most influential words for the topic.
+    '''
```
Comment (jinjahninjah) [inline app/document_frequency.py:43]: Just swap out these single quotes for double quotes 👍

**Review by jinjahninjah — CHANGES_REQUESTED** [review]
Just swap out the single quotes for double quotes and fix the PR description boxes and all will be well 😄

**Comment by BrandonSmith710** [comment]
> Just swap out the single quotes for double quotes and fix the PR description boxes and all will be well 😄

Sure thing, thanks.

**Review by jinjahninjah — APPROVED** [review]
:shipit:

**Pull request closed without merging**